Update regional_communities.md (#2)
near-ndc/gwg: specs/regional_communities.md

---

> Spec Nbr: 2

Reviewer: ```suggestion
Spec ID: 2
```

Most communities form around some intersection of geography, language and vertical constellation / interest. But they are not getting any help in finding their place in these dimensions. Similarly, new users looking for a community to join have a hard time finding the right community in the right intersection.

There is no natural evoution provided for communities that grow "too large" or too diverse. If they want to split then the only option is to create a new sibling- or child-community, again from scratch. Similarly, there is no natural way of merging two communities without having to agree to delete one (with all it's history) and ask the members each to manually move to the new community.

## Scope and Requirements

Describe:
- Expected functionality.
- Potential security issues and requirements.
- User profile description. Who is the user? How she/he is expected to us it?
- Requirements, assumptions and dependencies (internal and external).
- Outline identified potential problems and roadblocks related to building the solution.

> - Optionally: Budget limit.

Author: I guess I didn't make a full spec yet. More work to be done here.

> ### Scope

Author: Are you asking me to add text to describe feature scope or if we should add a new heading to the template? In any case, not sure what "feature scope" is or how it's different from features and scope separately. Maybe you can give some examples?

<!-- Define the scope and potential phases of the widget -->
In scope
- All geographies
- All languages
- All verticals
- All human users interested in joining, forming or managing communities of likeminded

Typical resons to come together in a community:
- Networking
- Learning from each other, discussing
- Support and help each other
- Find new friends, build meaningful relationship

What is NOT within the scope of regional communities? Groups that are not bulding lasting relationships, groups where the only purpose is to facilitate one-time or temporary transactions.
- Buy and sell groups
- Job listing / job search

> - Gig economy

Author: The focus should be on network relationships. Gig economy might create lasting relationships but they are peer-to-peer. This app is to facilitate open networks, not private convos.

### Backwards Compatibility
All proposals that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The doc must explain how to deal with these incompatibilities or the set of requirements to for the backward compatibility to be discussed in a submission. Submissions without a sufficient backwards compatibility treatise may be rejected outright.

## Phases
<!-- Do the project have multiple phases? Identify a high-level summary of each phase. -->

### Phase 1: MVP
- The app owner can create global parent communities for each Constellation
- Users can find their community, assign them selves membership permissionlessly (similar to "follow")
- Members can start posting messages to the Community board.
- Members can perform a sibling split along one of the three dimensions
- Members can perform a child split along one of the three dimensions, 
- Member executing the split will become the owner of the new Community

### Phase 2: Admin Functions
- Admins can assign other admins
- Users can request core memberships
- Admins can approve core memberships

> - Admins can warn, silence and ban users

Author: With silence they can still see what's going on. With ban they cannot (if it's a closed membership community). For open communities it might be the same. 

Yes, we should consider an appeal process and an un-ban feature.

### Phase 3: Enable voting
- Members can request a vote on splitting the Community
- Members can launch simple EasyPolls

> - Members can vote in a new core user

Author: How does this work now? Do I update the file in the branch and re-PR? Or do we merge first and then I do a new fresh PR?

- Members can vote to warn / silence / ban an offending user

## Use Cases

> ## Actors

Author: Really? I saw use cases as made up of 1) actors, 2) actions and 3) the actor-actions matrix.

> ## Actors

Reviewer: This is not what I meant. Use Cases section has the same level as Actors. The intention is to have Actors, Actions etc... under Use Cases section.

<!-- List all collections that will use the widget. -->
- Members
   - Admin members
   - Core members
   - Home members
   - Hangaround members
- Non-members
   - Non-member humans
   - Non-member users

## Actions
<!-- List the actions each collective will take individually. -->

### Find and Join a Community
Finding a community aligned with my interests should be very easy. It's a core concept of this widget and serves to get people into open and collaborative communities rather than creating competing factions.

Searching for communities should be possible through a few different ways
- Search by map. Zoom into a map and update the list of available communities accordingly
- Seach-as-you-type.
   - Search by fuzzy search of geographies. Start typing in a city, country or region and all available options are shown. 
   - Search by fuzzy search of constellations. Start typing in a constellation and all available options are shown.
- Select you language, which updates the available communities. Default language is English.
- Any combination of the above will further narrow the search

### Approve a new Community member

> Approvind a new community member could be as simple as a "follow back" by the owner of the community or someone who has been given the keys to control the community account. 

Reviewer: ```suggestion
Approvind a new community member could be as simple as a "follow back" by the owner of the community or someone who has admin rights to the community account. 
```

> It is recommended that only human-verified accounts be allowed to become members, as a minimum requirement.

Reviewer: I think we may get a push back of not allowing any account to become a member. I propose:
* community may optionally require only human-verified accounts for members. 
So, communities will have a choice if they are open for any accounts or only humans.

Author: Sure. I was going to leave it up to the developer, hackathon style, but we can be more prescriptive too.

> A drawback of this approach is that there is no vote. Any single admin can follow or unfollow, which may be too arbitrary for some communities. Another drawback is that the "follow" feature would no longer be available to use for its originally intended purpose. A more structured approach to membership may have to be contemplated. Optins include:

Reviewer: what is "follow"?

Author: You can follow a near social account today already much like you can follow someone on Facebook or Instagram. It has a meaning which is different from joining a community as an active member.

- user follows the Community and automatically becomes a member as long as some criteria are filled (e.g. is human verified and accepts the code of conduct) => Preferred
- user applies for membership, followed by a majority vote to approve the application
- membership is open to anyone without approval

> - membership is open to anyone, with the restriction that they can be a full member of one and only one community

Reviewer: Why such restrictive requirement? I may be a member of Swahili speakers community, and also member of ReFi community.

Author: Not restrictive. These are different membership types, all of which could be available in one community. Most communities would allow multiple types of memberships (all enabled by default). Of course you can be member of many communities. 

I was saying there could be one membership type whereby you have to pick your one and only "home" community. But maybe this is not a membership type as much as it is a flag.

This author recommends a combination, whereby there are four types of membership:

> 1. Home Membership - each user can define their one and only "home" membership = the community that is most closely aligned with them. This can be done without approval from the community, but each user can only select exactly one home community (or not make the selection).

Reviewer: ditto, home can have many dimensions.

Author: Sure. But if home was exclusive, then you could get extra benefits if you made this community your home. Maybe you assign three: one geography home, one language, and one tag that's your fav tag. Something that shows your dedication to a cause. "You can't be an admin in our community unless it aligns with at least two of your three home selections"

Anyways. It was a thought. Maybe a bad idea.

2. Hangaroud Membership - any user can be a hangaround. This feature is similar to a standard "follow" and should be the default membership for most communities.
3. Core Member - the user has been approved in some way (by community vote or admin vote) to be a core member of the community. This option should not be a default option, but could be a possibility to enable for some communities that desire this option.
4. Admin Member - the user has admin rights for the group

It is this authors recommendation that commuinties be open for anyone to join and that applications are "accepted by default". If limitations are desired then it's better to use the "silence the member" feature than to go to "accepted pending approval" which signals a closed or restricted society.

### Split a Community
Instead of creating a Community from scratch, splitting an existing Community into two should be the default way of creating new communities. This is a core feature of the Community widget design.

When splitting a community, the key question to ask the user is this: "Along which dimension do you want the split to occur: Geography, Language, or Constellation?"

> - If splitting along the Geography dimension then ask also if it's a sibling-split of child-split. 

Reviewer: who and how verifies the geography dimension? full KYC may be tricky and will require users to pay.

Author: Either first one to suggest a new geo-split gets to do it. Or we assign a panel to curate and approve it. I like the idea of free assignment, combined with letting communities choosing the parent they most align with, for a naturally evolving meta-hierarchy.

- If splitting along the Constellations dimension then
   - if there is already a constellation picked, then simply ask what constellation the new Community should serve and it becomes a sibling Community. Example: [Near Somalia DeFi] splits into [Near Somalia DeFi] + [Near Somalie Gaming]
   - if there was no prior constellation picked, then ask which Constellations the new Community should serve and becomes a child Community. Example: [Near Uganda] splits into [Near Uganda] + [Near Uganda NFT]
- The split has to be along at one and only one of the dimensions. The split cannot proceed unless there is an actual change proposed, one that will make the new Community different by design from the original Community from which it splits.

Any member can request a split, it must not be approved by the admins. 
- The members voting should be asked a few questions
   - Do you approve of this split proposal?
   - If so, would you want to become a member of the new Community? Would you voluenteer to also become and admin?
   - If so, would you also want to stay a member of the current community?

> - It is this author's opinion that a minimum of three to five other members should approve the split and also wish to become a member of the new Community for it to take effect

Reviewer: so 3 users are enough to setup a new community?

- Once sufficiently approved, then the split should immediately be executed. It should not require admin involvement or approval, because we want the split to be a natural, organic and permissionless occurence.

> The user requesting the split trigger an automatic vote to approve the split. If the request is approved then the original requestor will also be the first leader of the new community by default. 

Author: Maybe for a version 2, sure. 
I was going for something quick and easy to build here.

> After splitting the community, then each member should be given a choice as to which of the resulting communities they wish to belong to, or if they wish to be a member of both.

Reviewer: how this should be executed? How a user will know about it? Some sticky notification is required with option to dismiss it.

Author: The vote could have two questions: 1) approve the split? 2) assuming the split is approved, which resulting entity/entities would you want to belong to?

### Merge two Communities
We don't anticipate this to happen often, but it could be the case that two communities have shrinking membership and would like to merge to increase the total membership in one Community.

A merge proposal should be approved by majority vote in both communities before it is allowed to happen.

> It's not easy to design rules for how the merge should be executed, based on the complexity of possible permutations of the pre-merge Communities. It seems plausible that merge power should not be given to regular users but could be given as an admin power to a dev / superuser. The alternative would be that merging not be supported, in which case the community will manually have to move from and abandon one community in favor of the other.

Reviewer: I think merging should be a "referendum" action

Author: Yeah. As in advisory, non-binding.

### Interact with my Community
Members should at minimum be able to post messages within the community which become publicly readable to anyone regardless of membership status, and where other members are allowed to comment on the message.

Messages posted on boards that I follow should show up in my personal feed in chronological order. But they don't show up in the feed of users who are not members and don't follow the community.

> Members or admins (depending on Community settings) should be able to post polls for the community, which only community members are allowed to vote in.

Author: No. 

They would be able to read the threads if they looked up the community and double clicked on it.

But only if they are members or follow the community does the thread show up in their own "feed" . This is how the "follow" feature works today on near.social.

As a member I should be able to see all other members listed in a sidebar, along with their membership / admin status.

### Ban a user from a Community
Banning is a strong power and should be used very restrictively. We suggest milder and more human forms of restricting users that are not following the code of conduct for a community. 
- Issue a warning for breach of code of conduct
- Temporarily silence the user for a day
- Temporarily silence the user for a week
- Silence the user until un-silenced (perhaps after taking some corrective action)
- Banning the user with chance of reapplying

> - Banning the user with chance of reapplying (permanently)

Reviewer: what's the difference between the 2 cases above?

Author: Missing a word "no" in the last one.

It is this author's opinion that silencing the user is a much preferred method over banning them. Silencing allows the user to keep following the activitiy of the community and still feel like they belong there. It creates less animosity and makes the world a nicer place. Consider not implementing the banning feature.

These measures can be designed such that each increasing level of banning must be preceeded by the milder forms before it can be implemented. E.g. must silence the user for a day before can silence them for a week.

### Manage a Community

> Communities should be open and welcoming, not closed for elite members only. We should design them such that members have as much powers as possible, and that admins are given as few god-like options as possible.

Reviewer: idea: no admin by default. Community  is DAO

Author: No admin: I like it. But how do we prevent potentially disastrous actions, such as a malicious member renaming the community and changing the purpose? A majority community vote is not a good option, as it's proven to be very hard to get quorum votes within loosely affiliated and large communities.

Author: Community is DAO: Needs definition of what a DAO is. If the admins are thought of as Council members then that's a DAO too.

Reviewer: > such as a malicious member renaming the community and changing the purpose? A majority community vote is not a good option, as it's proven to be very hard to get quorum votes within loosely affiliated and large communities.

Each community should have it's parameters to reach consensus (eg, min quorum, could be % or fixed number, eg 10)

| Ban a user from a Community | Yes         |          |          |          |
| Manage a Community | Yes         |          |          |          |
| Create a new Community |          |          | Yes         |          |

## Data Architecture
<!-- Describe the data architecture and principles -->

### Geographies
To prevent unmanageable complexity in the geography dimension, we propose that there should always be a "trunk" that sets the direction into narrower and narrower geographies, from which the language-branches and constellation-branches can expand. The trunk concept is the family of communities that don't have a constellation selected and that has English as their language. Only communities with these characteristics are allowed to create new "child" geographies. Conversely, any Commmunity that either has a non-english language or has narrowed down their Constellation will not be allowed to create new geographies. Instead they will be confined to only select pre-defined geographies from the trunk. 

We should create an initial set of "trunk" Communities
- Global
   - North America
      - Canada

>       - USA

Reviewer: ```suggestion
      - USA
      - Mexico
```

   - Asia
      - Indonesia
      - Japan
      - South Korea
      - Russia
      - Vietnam

>    - South America

Reviewer: if we have regions, then South America should be also split down into countries.

Author: I had a thought that we should just follow some conventional list of countries. But I don't like it. Too much risk that someone will complain about inclusion or exclusion (Gaza, Kurdistan, Cyprus, Taiwan, etc). It also gets hard to manage when you go inside countries. So I prefer an open and evolving hierarchy as proposed.

Reviewer: add more then less. I thing there will be more people happy if we add Taiwan then exclude it at all.

   - Africa
   - Europe
      - DACH
      - France
      - Portugal
      - Spain

>       - Ukraine

Reviewer: How about UK, Italy, Balkans, Central Europe?... 

Lot of missing places in Asia as well.

Reviewer: Overall, I think we should support every country, this split is too opinioinated.

Author: Of course we should support all, it should evolve as it wishes based on where people want their communities. The initial list was a minimum we should enable from day one, based on where Near communities are typically already strong.
